### PR TITLE
Fix Bug 1436145 - Green up our JS dependencies for Firefox 60

### PR DESCRIPTION
**Pull request by Mardak**
Most dependencies updated except the 3 that were failing from before: istanbul-instrumenter-loader,karma-sinon,loader-utils

## success
redux 3.6.0 -> 3.7.2
enzyme 3.2.0 -> 3.3.0
eslint 4.12.0 -> 4.17.0
enzyme-adapter-react-16 1.1.0 -> 1.1.1
eslint-plugin-no-unsanitized 2.0.1 -> 2.0.2
eslint-plugin-react 7.5.1 -> 7.6.1
github 12.1.0 -> 13.1.1
karma 1.7.1 -> 2.0.0
karma-firefox-launcher 1.0.1 -> 1.1.0
karma-webpack 2.0.6 -> 2.0.9
mocha 4.0.1 -> 5.0.0
shelljs 0.7.8 -> 0.8.1
simple-git 1.83.0 -> 1.89.0
sinon 4.1.2 -> 4.2.2
webpack 3.8.1 -> 3.10.0

## fail
istanbul-instrumenter-loader 0.2.0 -> 3.0.0
karma-sinon 1.0.5 -> 1.8.0
loader-utils 0.2.16 -> 1.1.0

## major version updates
github (now octokit/rest):
- https://github.com/octokit/rest.js/releases/tag/v13.0.0
- https://github.com/octokit/rest.js/releases/tag/v14.0.0

karma
- https://github.com/karma-runner/karma/releases/tag/v2.0.0

mocha
- https://github.com/mochajs/mocha/releases/tag/v5.0.0

### CODE REVIEW

**Comment by Mardak** [comment]
@k88hudson there is an update to Redux 3.7.2 and we're 3.6.0:
https://www.npmjs.com/package/redux

It's been available for 7 months now. Not sure if we just do that separately as there's a Redux.jsm that has local changes and needs to be manually updated too?

**Comment by k88hudson** [comment]
@Mardak I would do it separately, but let's make sure the manual update is in the same commit as the package.json commit, but yeah

File: bin/test-merges.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 const Task = require("co-task");
 const process = require("process");
 const path = require("path");
-const GitHubApi = require("github");
+const GitHubApi = require("@octokit/rest");
 const shelljs = require("shelljs");
 const child_process = require("child_process");
 const github = new GitHubApi({debug: false});
```
Comment (dmose) [inline bin/test-merges.js]: According to the relnotes for v13, support for "debug" was taken out then, so you can probably just remove the parameter from this line entirely.

**Comment by as-pine-proxy** [comment]
[Treeherder: c04ea46f02a25cbcce230b362f196f6f852ca2a9](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c04ea46f02a25cbcce230b362f196f6f852ca2a9)

**Pull request merged**